scopes/component/component-descriptor/component-descriptor.ts

```diff
@@ -0,0 +1,66 @@
+import { ComponentID } from '@teambit/component';
+import { AspectMapProps, AspectMap } from './aspect-map';
+
+export type ComponentDescriptorProps = {
+  /**
+   * serialized component ID.
+   */
+  id: {
+    name: string;
+    scope: string;
+    version?: string;
+  };
+  /**
+   * aspects map data
+   */
+  aspectMap?: AspectMapProps;
+};
+
+export class ComponentDescriptor {
+  constructor(
+    /**
+     *  Component ID
+     */
+    readonly id: ComponentID,
+    /**
+     *  aspect map
+     */
+    readonly aspectMap: AspectMap
+  ) {}
+
+  get state() {
+    return this.aspectMap;
```

Comment:
This is not the only stuff which is part of the state.
What about the source code of the component for example.
See here - https://github.com/teambit/bit/blob/master/scopes/component/component/state.ts